spec/components/dashboard_component_spec.rb

```diff
@@ -241,4 +241,67 @@
       end
     end
   end
+
+  describe "#view_applicants" do
+    let(:organisation) { create(:school) }
+    let(:vacancies) { [vacancy] }
+    let(:vacancy) do
+      create(:vacancy, :published, organisations: [organisation])
+    end
+
+    before do
+      create(:job_application, :status_submitted, vacancy: vacancy)
+      # rubocop:disable RSpec/SubjectStub
+      allow(subject).to receive(:include_job_applications?).and_return(include_applications)
+      # rubocop:enable RSpec/SubjectStub
```

Comment:
You could set the context based on the `selected_type` (that is what dictates the inner behaviour of `#include_job_applications?`) and is already available to set as a `let` value in the subject initialisation.

```ruby
context "when the selected type is live" do
  let(:selected_type) { :live }

  it "renders the link to view applicants" do
  ...
end

context "when the selected type is draft" do
  let(:selected_type) { :draft }

  ...
end
```
This way, you don't have to define this "problematic" stub over the subject under test and also avoids stubbing a private method.

Reply:
Good idea, thanks! Done!